code/game/gamemodes/endgame/xmas/victorianstation.dm

```diff
@@ -0,0 +1,106 @@
+					T.ChangeTurf(/turf/simulated/wall/mineral/wood, tell_universe = 0)
+			for(var/obj/item/device/flashlight/F in A)
+				var/obj/item/candle/C = new /obj/item/candle(F.loc)
+				C.light(quiet = 1)
+				qdel(F)
+			for(var/obj/machinery/light/L in A)
+				var/turf/T = get_turf(L)
+				var/obj/structure/hanging_lantern/HL = new /obj/structure/hanging_lantern/dim(T)
+				HL.dir = L.dir
+				HL.update()
+				qdel(L)
+			for(var/obj/structure/closet/secure_closet/S in A)
+				switch(S.type)
+					if(/obj/structure/closet/secure_closet/captains)
+						new /obj/item/clothing/suit/wintercoat/captain(S)
+					if(/obj/structure/closet/secure_closet/hop2)
+						new /obj/item/clothing/suit/wintercoat/hop(S)
+					if(/obj/structure/closet/secure_closet/hos)
+						new /obj/item/clothing/suit/wintercoat/security/hos(S)
+					if(/obj/structure/closet/secure_closet/warden)
+						new /obj/item/clothing/suit/wintercoat/security/warden(S)
+					if(/obj/structure/closet/secure_closet/security)
+						new /obj/item/clothing/suit/wintercoat/security(S)
+					if(/obj/structure/closet/secure_closet/brig)
+						new /obj/item/clothing/suit/wintercoat/prisoner(S)
+						new /obj/item/clothing/suit/wintercoat/prisoner(S)
+					if(/obj/structure/closet/secure_closet/scientist)
+						new /obj/item/clothing/suit/wintercoat/science(S)
+						new /obj/item/clothing/suit/wintercoat/science(S)
+					if(/obj/structure/closet/secure_closet/RD)
+						new /obj/item/clothing/suit/wintercoat/science(S)
+					if(/obj/structure/closet/secure_closet/medical3 || /obj/structure/closet/secure_closet/paramedic)
+						new /obj/item/clothing/suit/wintercoat/medical(S)
+						new /obj/item/clothing/suit/wintercoat/medical(S)
+					if(/obj/structure/closet/secure_closet/CMO)
+						new /obj/item/clothing/suit/wintercoat/cmo(S)
+					if(/obj/structure/closet/secure_closet/engineering_chief)
+						new /obj/item/clothing/suit/wintercoat/ce(S)
+					if(/obj/structure/closet/secure_closet/engineering_personal || /obj/structure/closet/secure_closet/engineering_mechanic)
+						new /obj/item/clothing/suit/wintercoat/engineering(S)
+						new /obj/item/clothing/suit/wintercoat/engineering(S)
+					if(/obj/structure/closet/secure_closet/engineering_atmos)
+						new /obj/item/clothing/suit/wintercoat/engineering/atmos(S)
+						new /obj/item/clothing/suit/wintercoat/engineering/atmos(S)
+					else
+						if(prob(50))
+							new /obj/item/clothing/suit/wintercoat(S)
+							if(prob(80))
+								new /obj/item/clothing/suit/wintercoat(S)
+
+
+	var/area/christmas_bar = locate(/area/crew_quarters/bar)
+	if(christmas_bar)
+		var/list/turf/simulated/floor/valid = list()
+		//Loop through each floor in the supply drop area
+		for(var/turf/simulated/floor/F in christmas_bar)
+			if(!F.has_dense_content() && istype(F, /turf/simulated/floor/wood))
+				valid.Add(F)
+		new/obj/structure/snow_flora/tree/pine/xmas/vg/(pick(valid))
+
+
+	var/area/santadog = locate(/area/crew_quarters/hop)
```

Comment:
Why not just locate ian in world?

Reply:
Would that not be expensive?

Reply:
1.536   2.834    2.873 - checking corgi in world
1.540	2.694	2.731 - checking corgi in HoP office